### PR TITLE
SUMO-115364 Handle exceptions thrwon from kubeclient calls

### PR DESCRIPTION
**Pull request by yuting-liu**
Since we are using `timer_execute` for `start_monitor`, any exceptions thrown from the method will cause the timer detached.

We are going to add exceptions handling for all methods that have kubeclient calls in events plugin, including following methods:
```
start_watcher_thread
create_config_map
patch_config_map
initialize_resource_version
pull_resource_version
```
We will only rescue possible errors and then log the error so that we can know errors from which method they throw. For all other methods except for `start_watch_thread`, we don't want errors from kubeclient blocks the thread of watching, so we just log the error and ignore it. For `start_watcher_thread`, we need `timer_execute` continues working so exceptions will be swallowed and timer won't be detached in this case.

### CODE REVIEW

File: fluent-plugin-events/test/plugin/test_in_events.rb [hunk 1]
```diff
@@ -186,4 +186,65 @@ def connect_kubernetes_with_api_version(driver)
     
     driver.start_monitor
   end
+
+  sub_test_case 'exception from kubeclient will not interrupt execution of start monitor' do
```
Comment (maimaisie) [inline fluent-plugin-events/test/plugin/test_in_events.rb]: `create_config_map` and `initialize_resource_version` are not called within `start_monitor` right?

File: fluent-plugin-events/test/plugin/test_in_events.rb [hunk 2]
```diff
@@ -186,4 +186,65 @@ def connect_kubernetes_with_api_version(driver)
...
+    test 'exception from kubeclient call in create_config_map' do
+      config = %([])
+      driver = create_driver(config).instance
+      connect_kubernetes_with_api_version(driver)
+      driver.instance_variable_set(:@clients, @clients)
+      driver.instance_variable_set(:@last_recreated, 0)
```
Comment (maimaisie) [inline fluent-plugin-events/test/plugin/test_in_events.rb]: is this needed in test cases that don't call `start_monitor`? Also it looks like these 5 lines are being repeated in almost every test case, maybe we need to DRY or need something similar to the `beforeEach()` in scala

Comment (yuting-liu) [reply fluent-plugin-events/test/plugin/test_in_events.rb]: Good point. Going to refactor this.

File: fluent-plugin-events/test/plugin/test_in_events.rb (hunk 1 above)
Comment (yuting-liu) [reply fluent-plugin-events/test/plugin/test_in_events.rb]: Yes. Need to change this description then.

**Comment by yuting-liu** [comment]
Refactored the part of creating the test driver with configuration and instance variable setup into a method `configure_test_driver(config = %([]))`.

**Pull request merged**